<?xml version="1.0"?>
<launch>
    <!-- ============================ SETTINGS ============================= -->

    <!-- Node frequency: the frequency [Hz] which rate.sleep() function will
    attempt to maintain in the main_loop. -->
    <arg 
    name="node_frequency" 
    default="100"/>

    <!-- Controller side: "right" or "left". -->
    <arg 
    name="controller_side" 
    default="right"/>

    <!-- Max forward speed [rad/s]: default is 0.5. -->
    <arg 
    name="max_forward_speed" 
    default="0.25"/>

    <!-- Max angular speed [rad/s]: default is 0.5. -->
    <arg 
    name="max_angular_speed" 
    default="0.5"/>

    <!-- If any of dependency or optional subfiles should not be launched (or
    are launched separately) set corresponding argument to "false". -->
    <arg 
    name="launch_controller_feedback" 
    default="true"/>

    <arg 
    name="launch_velocity_control" 
    default="true"/>

    <!-- ======================== END OF SETTINGS ========================== -->


    <!-- Oculus controller feedback (dependency): -->
    <group if="$(arg launch_controller_feedback)">
        <include 
        file="$(find oculus_ros)/launch/controller_feedback.launch">
            <arg 
            name="controller_side" 
            value="$(arg controller_side)"/>
            <arg 
            name="enable_joystick_deadzones" 
            value="true"/>
        </include>
    </group>

    <!-- Mobile base velocity control (optional): -->
    <group if="$(arg launch_velocity_control)">
        <include 
        file="$(find hirolab_fetch_freight100)/launch/velocity_control.launch">
            <arg 
            name="max_forward_speed" 
            value="$(arg max_forward_speed)"/>
            <arg
            name="max_angular_speed" 
            value="$(arg max_angular_speed)"/>
        </include>
    </group>

    <!-- Oculus mapping: -->
    <node 
    pkg="hirolab_fetch_freight100" 
    type="oculus_mobile_base_mapping.py" 
    name="oculus_mobile_base_mapping" 
    output="screen">
        <param 
        name="node_frequency" 
        value="$(arg node_frequency)"/>
        <param 
        name="controller_side" 
        value="$(arg controller_side)"/>
        <param 
        name="max_forward_speed" 
        value="$(arg max_forward_speed)"/>
        <param 
        name="max_angular_speed" 
        value="$(arg max_angular_speed)"/>
    </node>
</launch>